client/src/pages/MainPage.jsx

```diff
@@ -8,6 +8,8 @@ import "../styles/MainPage.css";
 
 const MainPage = () => {
 
+    const defaultEvents = useRef([]);
```

Comment:
Can you provide a bit more details on why you're relying on useRef here? Perhaps context or state would be a better tool here.